client/src/components/common/boost/BoostCalibration.tsx

```diff
@@ -0,0 +1,98 @@
+import React , {useState, useCallback} from 'react';
+import {Button, Card, Form, Col} from 'react-bootstrap';
+
+export interface BoostCalibrationProps {
+    onSet: (calibValue: number) => void;
+    onReset: () => void;
+    distTravelled: number;
+  }
+
+/**
+ * Boost calibration card, displays the calibrated distance and the 
+ * distance travelled by the bike while also allowing user to set the 
+ * calibrated distance
+ * 
+ * @returns {React.Component} Component
+ */
+export default function BoostCalibration({
+    onSet, onReset, distTravelled
+}: BoostCalibrationProps) {
+    // Controls the feedback form
+    const [validated, setValidation] = useState(false);
+    
+    const [calibValue, setCalibValue] = useState(0);
+    // Holds the difference between the current distance and the calibrated distance
+    const [calibDiff, setCalibDiff] = useState(0);
+
+    const handleCalibrationChange = useCallback(
+        (event) => {
+        setCalibValue(event.target.value);
+        },
+        [setCalibValue],
+    );
+
+  const handleSubmit = useCallback(
+      (event) => {
+        const form = event.currentTarget;
+        if (form.checkValidity() === false) {
+        event.stopPropagation();
+        }
+        else {
+            onSet(calibValue);
+            setCalibDiff(calibValue-distTravelled);
+        }
+        event.preventDefault();
+        setValidation(true);
+    },
+    [calibValue, onSet, setValidation]);
```

Comment:
I'm getting the following linting warning:
`React Hook useCallback has a missing dependency: 'distTravelled'. Either include it or remove the dependency array.`
Should be an easy fix :)